src/app/destinyTrackerApi/dtr-service-helper.ts

```diff
@@ -29,27 +40,37 @@ export function dtrFetch(url: string, body: object) {
   }
 
   return new Promise((resolve, reject) => {
-    circuitBreaker.run(
-      (success, failure) => {
-        Promise.resolve(fetch(request, { signal }))
-          .then((r) => {
-            if (controller) {
-              clearTimeout(timer);
-            }
-            return r;
-          })
-          .then(
-            (v) => {
-              success();
-              resolve(v);
-            },
-            (e) => {
-              failure();
-              reject(e);
-            }
-          );
-      },
-      () => reject(new Error('Circuit breaker open'))
-    );
+    if (fiveOhThreeCaughtRecently()) {
+      reject(new Error('Waiting due to HTTP 503'));
+    } else {
+      circuitBreaker.run(
+        (success, failure) => {
+          Promise.resolve(fetch(request, { signal }))
+            .then((r) => {
+              if (r.status === 503) {
+                lastFiveOhThreeCaught = new Date();
+                failure();
+                reject(new Error('HTTP 503 returned'));
```

Comment:
Should probably add these to the sentry exception list.

https://github.com/DestinyItemManager/DIM/blob/61de8af1184d759b04da77d21049afa6bd46511d/src/app/utils/exceptions.ts

Reply:
I added the two messages to the exception list. Thanks for catching that!